transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java

```diff
@@ -109,7 +109,8 @@ int fd(int index) {
 
     private int getInt(int index, int offset) {
         if (PlatformDependent.hasUnsafe()) {
-            return PlatformDependent.getInt(memoryAddress + index * EPOLL_EVENT_SIZE + offset);
+            long n = index * EPOLL_EVENT_SIZE;
+            return PlatformDependent.getInt(memoryAddress + n + offset);
```

Comment:
why was this changed ?

Reply:
LGTM reports a potential integer overflow for `index * EPOL_EVENT_SIZE`:

https://lgtm.com/projects/g/netty/netty/snapshot/46cad8a5db67884faf2f41f99ed3a74ffa3735f9/files/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java?sort=name&dir=ASC&mode=heatmap#x3b93cf5f22ba2792:1

I should have convert `index` and the constant to `long` as well. Now it's fixed.